app/services/TextBuilder.scala

```diff
@@ -28,13 +28,17 @@ object TextBuilder {
   private val answerHintPattern: Regex = """\[hint:([^\]]+)\]""".r
 
   private object Placeholders { // All the placeholder matching in one place
-    val plregex: Regex = s"\\[label:([A-Za-z0-9\\s\\-_]+)\\]|\\[bold:([^\\]]+)\\]|\\[link(-same|-tab)?:([^\\]]+?):(\\d+|${Process.StartStanzaId}|https?:[a-zA-Z0-9\\/\\.\\-\\?_\\.=&]+)\\]".r
+    val labelPattern = "\\[label:([A-Za-z0-9\\s\\-_]+)\\]"
+    val boldPattern = "\\[bold:([^\\]]+)\\]"
+    val linkPattern = s"\\[(button|link)(-same|-tab)?:([^\\]]+?):(\\d+|${Process.StartStanzaId}|https?:[a-zA-Z0-9\\/\\.\\-\\?_\\.=&]+)\\]"
+    val plregex: Regex = s"${labelPattern}|${boldPattern}|${linkPattern}".r
     def labelNameOpt(m: Match): Option[String] = Option(m.group(1))
     def boldTextOpt(m: Match): Option[String] = Option(m.group(2))
-    def linkTypeOpt(m: Match): Option[String] = Option(m.group(3))
-    def linkText(m: Match): String = m.group(4)
+    def buttonOrLink(m: Match): Option[String] = Option(m.group(3))
```

Comment:
I think we need to test these need place holder types in the bullet point list builder tests.